### PR TITLE
feat(forms): improve error message when formGroup directive receives a FormArray instance

### LINKED ISSUE #14471: Throw when formGroup directive receives a FormArray instance
Currently we only throw when the formGroup directive is missing its input. Given that it doesn't accept FormArrays or other types, we should be throwing a helpful error (probably the existing `ReactiveErrors.missingForm`).

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #14471

### CODE REVIEW

**Comment by ericmartinezr** [comment]
Super cool, thanks for this!!

PS : You were unbanned from gitter :P

**Comment by DzmitryShylovich** [comment]
@ericmartinezr I don't need it anymore. Dobby is free :P

File: modules/@angular/forms/src/directives/reactive_directives/form_group_directive.ts [hunk 1]
```diff
@@ -162,23 +160,26 @@ export class FormGroupDirective extends ControlContainer implements Form,
...
     this.form.asyncValidator = Validators.composeAsync([this.form.asyncValidator, async]);
   }
 
-  private _checkFormPresent() {
+  private _checkFormPresentAndValid(): void {
     if (!this.form) {
       ReactiveErrors.missingFormException();
     }
+    if (!(this.form instanceof FormGroup)) {
+      ReactiveErrors.expectFormGroupException((<any>this.form).constructor.name);
```
Comment (DzmitryShylovich) [inline modules/@angular/forms/src/directives/reactive_directives/form_group_directive.ts]: without `any` it throws [Property does not exist on type 'never'](https://github.com/Microsoft/TypeScript/issues/12083)

Comment (devoto13) [inline modules/@angular/forms/src/directives/reactive_directives/form_group_directive.ts]: Not sure if it's a real issue, as error is intended for development mode. But isn't `Function.name` property affected by name mangling after minification?

Comment (DzmitryShylovich) [inline modules/@angular/forms/src/directives/reactive_directives/form_group_directive.ts]: > But isn't Function.name property affected by name mangling after minification?

it is, but a don't think it's an issue. As you said it's only intended for development.

**Comment by kara** [comment]
Closing as stale

**Pull request closed without merging**